.github/workflows/release.yml

```diff
@@ -4,36 +4,81 @@
 
 # This workflow will publish a Java project with Gradle
 # For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-gradle
+# Inspired by https://github.com/dzikoysk/reposilite/blob/main/.github/workflows/publish-release.yml
 
-name: Publish Release to Maven Central
+name: Publish Release
 
-on:
-  release:
-    types: [published]
+on: workflow_dispatch
 
 jobs:
-  build:
+
+  github:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+
+      - name: Checkout repository
+        uses: actions/checkout@v3 # TODO: with... bot?
+
+      - name: Fetch git tags # Required for axion-release-plugin
+        run: git fetch --tags
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: adopt
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1.1.0
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2.7.0
+
+      - name: Release new version
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+          ./gradlew release
+
+      - name: Get current version
+        id: version
+        run: echo "::set-output name=version::$(./gradlew -q -Prelease.quiet currentVersion)"
```

Comment:
What does this?

Reply:
Added a comment before this step to explain its purpose & functionality. Is this sufficient?

Reply:
That's a good starting point for the reason, but there should also be some info about the concrete syntax (`::set-output name=version::`). If you have a link which describes this, a link with a hint that it describes the syntax is sufficient.